File: lab2/menu.c
#include <stdio.h>
#include <stdlib.h>

void help();
void hello();
void quit();
void sum();
void compare();
void echo();
void ls();
void mkdir();

int main()
{
    char cmd[128];
	printf("This system has the cmd below:\n");
	printf("hello\n");
        printf("help\n");
        printf("quit\n");
        printf("sum\n");
        printf("compare\n");
        printf("echo\n");
        printf("ls\n");
        printf("mkdir\n");

    while(1)
    {
        printf("\n$:");
        scanf("%s", cmd);
        if (strcmp(cmd, "hello") == 0)
        {
            hello();
	}
        else if(strcmp(cmd, "help") == 0)
        {
            help();
        }
        else if(strcmp(cmd, "quit") == 0)
        {
	    quit();
        }
        else if(strcmp(cmd, "sum") == 0)
        {
            sum();
        }
        else if(strcmp(cmd, "compare") == 0)
        {
	    compare();
        }
        else if(strcmp(cmd, "echo") == 0)
        {
            echo();
        }
        else if(strcmp(cmd, "ls") == 0)
        {
            system(cmd);
        }
        else if(strcmp(cmd, "mkdir") == 0)
        {
            mkdir();
        }
        else
        {
            printf("Warning: WORRY COMMAND!\n");
        }
   }
}

void hello()
{
    printf("Hello, Wlecome to this menu~\n");
}

void help()
{
    printf("Need help? I will try my best to help you!");
}

void quit()
{
    exit(0);
}

void sum()
{
    int a = 0;
    int b = 0;

    printf("Input two number:\n");
    scanf("%d, %d", &a, &b);
    printf("a+b = %d\n", a+b);
}

void compare()
{
    int c = 0;
    int d = 0;
    int bigger = 0;

    printf("Input two numbers:\n");
    scanf("%d, %d", &c, &d);
    bigger=(c>d?c:d);
    printf("The bigger one is: %d", bigger);
}

void echo()
{
    char buf[128];

    printf("Hello, strat your performance pls.\n");
    fgetc(stdin);
	fputs(fgets(buf,128,stdin),stdout);
}

void mkdir()
{
    printf("You want make a new dir? Sorry, i have no access to it.\n");
}

File: lab2/menu.go
package main
import 
(
	"fmt"
	"strings"
	"os/exec"
	"os"
	"bufio"
)

var cmd string

func main()  {
	fmt.Println("This system has cmd below:")
	fmt.Println("hello")
	fmt.Println("help")
	fmt.Println("quit")
	fmt.Println("sum")
	fmt.Println("compare")
	fmt.Println("echo")
	fmt.Println("ls")
	fmt.Println("mkdir")
	for true {
		fmt.Printf("\n$:")
        fmt.Scanln(&cmd)
	    if strings.EqualFold(cmd, "hello") {
	    	hello()
	    }else if strings.EqualFold(cmd, "help") {
	    	help()
	    }else if strings.EqualFold(cmd, "quit") {
	    	quit()
	    }else if strings.EqualFold(cmd, "sum") {
	    	sum()
	    }else if strings.EqualFold(cmd, "compare") {
	    	compare()
	    }else if strings.EqualFold(cmd, "echo") {
	    	echo()
	    }else if strings.EqualFold(cmd, "ls") {
	    	ls()
	    }else if strings.EqualFold(cmd, "mkdir") {
	    	mkdir()
	    }else {
	    	fmt.Println("Warning: WORRY COMMAND!!")
	    }
	}
	
}
func hello() {
	fmt.Println("Hello, Wlecome to this menu~")
}
func help() {
	fmt.Println("Need help? I will try my best to help you!")
}
func quit() {
	os.Exit(1)
}
func sum() {
	a:=0
	b:=0

	fmt.Println("Input two numbers:")
	fmt.Scanf("%d %d", &a, &b)
	fmt.Println("Sum of two numbers: ",a+b)	
}
func compare() {
	c:=0
	d:=0
	bigger:=0

    fmt.Println("Input two numbers:")
    fmt.Scanf("%d %d", &c, &d)
    if c>d {
    	bigger = c
    } else {
    	bigger = d
    }
    fmt.Println("The bigger one is: ", bigger)
}
func echo() {
	buf := bufio.NewReader(os.Stdin)
	
	line, err := buf.ReadString('\n')
	if err != nil{
		fmt.Println(err)
	}
	fmt.Println(line)
}
func mkdir() {
	fmt.Println("You want make a new dir? Sorry, i have no access to it.")
}
func ls() {
	cm := exec.Command("ls") 
    out, err := cm.CombinedOutput()  
    if err != nil {  
        fmt.Println(err)  
    }  
    fmt.Println(string(out))  
}